### PR TITLE
Adds table of contents to the README

### PR DESCRIPTION
**Pull request by tterranigma**
- [x] I have reviewed the [Contributor Guidelines][contributor].
- [x] I have reviewed the [Code of Conduct][coc] and agree to abide by it

[contributor]: CONTRIBUTING.md
[coc]:         CODE_OF_CONDUCT.md

### CODE REVIEW

**Comment by sublimino** [comment]
Made some standardisation changes ("BATS" -> "Bats"), checked against a TOC generated by https://github.com/jonschlinkert/markdown-toc and LGTM.

Would appreciate a merge from @mbland or @btamayo as I've added changes to this PR.

File: README.md [hunk 1]
```diff
@@ -25,11 +47,10 @@ Doing it this way accomplishes two things:
 3. Allows the possibility of merging back to original, or merging from original if or when the need arises
 4. Prevents lock-out by giving administrative access to more than one person, increases transferability
 
-## Misc
+### Misc
```
Comment (mbland) [inline README.md]: Should probably rename this to `Contact us`. (And I still have to learn how to securely sign into freenode.)

File: README.md [hunk 2]
```diff
@@ -1,15 +1,37 @@
-## BATS-core: Bash Automated Testing System (2017)
+# Bats-core: Bash Automated Testing System (2018)
 
 [![Build Status](https://travis-ci.org/bats-core/bats-core.svg?branch=master)](https://travis-ci.org/bats-core/bats-core)
 
-### Background:
+### Table of contents
```
Comment (mbland) [inline README.md]: This should be a `##` to be at the same level as `## Background`.

Comment (tterranigma) [reply README.md]: Silly me...

File: README.md (hunk 1 above)
Comment (tterranigma) [reply README.md]: Renamed it. To me, it would be better if this section was merged with the `Support` section at some point.

**Pull request merged**